undertow/src/test/java/org/wildfly/extension/undertow/UndertowServiceTestCase.java

```diff
@@ -0,0 +1,140 @@
+        final UndertowService undertowService = undertowServiceService.getValue();
+        assertNotNull(undertowService);
+        return undertowService;
+    }
+
+    @Test
+    public void testUndefinedAttributes() throws Exception {
+        final UndertowService undertowService = load("undertow-service-undefined-attributes.xml");
+        assertEquals(NODE_NAME, undertowService.getInstanceId());
+        assertFalse(undertowService.isStatisticsEnabled());
+        assertEquals(DEFAULT_SERVER, undertowService.getDefaultServer());
+        assertEquals(DEFAULT_SERVLET_CONTAINER, undertowService.getDefaultContainer());
+        assertEquals(DEFAULT_VIRTUAL_HOST, undertowService.getDefaultVirtualHost());
+    }
+
+    @Test
+    public void testDefinedDefaultAttributes1() throws Exception {
+        final UndertowService undertowService = load("undertow-service-defined-default-attributes1.xml");
+        assertEquals(NODE_NAME, undertowService.getInstanceId());
+        assertFalse(undertowService.isStatisticsEnabled());
+        assertEquals(DEFAULT_SERVER, undertowService.getDefaultServer());
+        assertEquals(DEFAULT_SERVLET_CONTAINER, undertowService.getDefaultContainer());
+        assertEquals(DEFAULT_VIRTUAL_HOST, undertowService.getDefaultVirtualHost());
+    }
+
+    @Test
+    public void testDefinedDefaultAttributes2() throws Exception {
+        final UndertowService undertowService = load("undertow-service-defined-default-attributes2.xml");
+        assertEquals(NODE_NAME, undertowService.getInstanceId());
+        assertFalse(undertowService.isStatisticsEnabled());
+        assertEquals(DEFAULT_SERVER, undertowService.getDefaultServer());
+        assertEquals(DEFAULT_SERVLET_CONTAINER, undertowService.getDefaultContainer());
+        assertEquals(DEFAULT_VIRTUAL_HOST, undertowService.getDefaultVirtualHost());
+    }
+
+    @Test
+    public void testDefinedAttributes() throws Exception {
+        final UndertowService undertowService = load("undertow-service-defined-attributes.xml");
+        assertEquals(NODE_NAME + "-undertow", undertowService.getInstanceId());
+        assertTrue(undertowService.isStatisticsEnabled());
+        assertEquals("undertow-server", undertowService.getDefaultServer());
+        assertEquals("servlet-container", undertowService.getDefaultContainer());
+        assertEquals("virtual-host", undertowService.getDefaultVirtualHost());
+    }
+
+    @Test
+    public void testUuidInstanceId() throws Exception {
+        final UndertowService undertowService = load("undertow-service-uuid-instance-id-attribute.xml");
+        assertNotNull(UUID.fromString(undertowService.getInstanceId()));
+        assertFalse(undertowService.isStatisticsEnabled());
+        assertEquals("undertow-server", undertowService.getDefaultServer());
+        assertEquals(DEFAULT_SERVLET_CONTAINER, undertowService.getDefaultContainer());
+        assertEquals("virtual-host3", undertowService.getDefaultVirtualHost());
+    }
```

Comment:
@fl4via looking at default `standalone-ha.xml` shipped e.g. with EAP 7.3 I see that element `application-security-domains` is not defined by default; in all the files undertow-service-*.xml used in the tests that element is always defined: would it make sense to add one more tests with no `application-security-domains`?

Reply:
certainly, @tommaso-borgato , I'll make that change